base/bitarray.jl

```diff
@@ -1272,29 +1272,45 @@ end
 (>>)(B::BitVector, i::Int32) = B >>> i
 (>>)(B::BitVector, i::Integer) = B >>> i
 
+function rol!(dest::BitVector, src::BitVector, i::Integer)
+    if length(dest) == length(src)
```

Comment:
The check should throw an error in case it's false, rather then continuing silently